adcm_aio_client/core/config/_base.py

```diff
@@ -0,0 +1,346 @@
+        value = self._current_config_level[name]
+        spec = self._get_parameter_spec(name=name)
+
+        is_group = isinstance(value, dict) and self._parameter_is_group(parameter_spec=spec)
+
+        if is_group:
+            # value for groups isn't copied,
+            # because there isn't public interface for accessing it + they can be quite huge
+            level_data = value
+            previous_levels = (*self._previous_levels, name)
+
+            is_activatable = (spec["adcmMeta"].get("activation") or {}).get("isAllowChange")
+            if is_activatable:
+                return ActivatableGroupWrapper(
+                    config_level_data=level_data, spec=spec, callbacks=self._callbacks, previous_levels=previous_levels
+                )
+
+            return RegularGroupWrapper(
+                config_level_data=level_data, spec=spec, callbacks=self._callbacks, previous_levels=previous_levels
+            )
+
+        if isinstance(value, (dict, list)):
+            # simple failsafe for direct `value` edit
+            value = deepcopy(value)
+
+        set_value_callback = partial(self._callbacks.set_value, (*self._previous_levels, name))
+
+        return ValueWrapper(value=value, set_value_callback=set_value_callback)
+
+
+type ParameterWrapper = ValueWrapper | RegularGroupWrapper | ActivatableGroupWrapper
+
+
+class RegularGroupWrapper(_ParametersGroup):
+    def __init__(
+        self: Self, config_level_data: dict, spec: dict, callbacks: Callbacks, previous_levels: LevelNames
+    ) -> None:
+        super().__init__(spec=spec, callbacks=callbacks, previous_levels=previous_levels)
+        self._data = config_level_data
+
+    @property
+    def _current_config_level(self: Self) -> dict:
+        return self._data
+
+
+class ActivatableGroupWrapper(RegularGroupWrapper):
+    def activate(self: Self) -> Self:
+        # silencing check, because protocol is position-based
+        self._callbacks.set_activation_attribute(self._previous_levels, True)  # noqa: FBT003
+        return self
+
+    def deactivate(self: Self) -> Self:
+        # silencing check, because protocol is position-based
+        self._callbacks.set_activation_attribute(self._previous_levels, False)  # noqa: FBT003
+        return self
+
+
+class EditableConfig(_ParametersGroup):
+    def __init__(
+        self: Self,
+        data: dict,
+        spec: dict,
+    ) -> None:
+        super().__init__(
+            spec=spec,
+            callbacks=Callbacks(
+                set_value=self._set_parameter_value, set_activation_attribute=self._set_activation_attribute
+            ),
+        )
+
+        self._initial_data = data
+        self._json_fields: set[tuple[ParameterName, ...]] = set()
+        self._convert_payload_formated_json_fields_inplace(self._spec, self._initial_data["config"], prefix=())
+
+        self._changed_data = None
+
+    def to_payload(self: Self) -> dict:
+        payload = deepcopy(self._current_data)
```

Comment:
purpose of this method (I see that it copy)? what problem solve this method?

Reply:
protection of internals from change mostly